.github/workflows/release.yaml

```
@@ -1,280 +1,358 @@
 name: Release
 
+# Only trigger this workflow when a tag is pushed in the format `vA.B.C`.
 on:
   push:
     tags:
-      - '*'
-
+      - "v[0-9]+.[0-9]+.[0-9]+*"
```

Do the dots have to be escaped, since it's Regex?

```suggestion
      - "v[0-9]+\.[0-9]+\.[0-9]+*"
```

---

Yes, the dot matches dot characters so this works but is overscoped

---

So it turns out that Github workflows does not use regex: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet.

`.` character has no special meaning, it's just literal `.`